File: solutions/cpp/isogram/1/isogram.cpp
#include "isogram.h"
#include <set>
#include <utility>

using namespace std;

namespace isogram
{
    bool is_isogram(string candidate)
    {
        set<char> chars;

        for (char c : candidate)
        {
            if (!isalpha(c))
                continue;

            char key = tolower(c);
            if (chars.find(key) == chars.end())
                chars.insert(key);
            else
                return false;
        }
        return true;
    }

} // namespace isogram

File: solutions/haskell/isogram/1/src/Isogram.hs
module Isogram (isIsogram) where

import Data.Char
import Data.List 

isIsogram :: String -> Bool
isIsogram text = 
  let lowerLetters = (map toLower) . (filter isLetter) $ text
  in length lowerLetters == (length . nub $ lowerLetters)